netsim/modules/srv6.yml

```diff
@@ -1,21 +1,52 @@
-# SRv6 default settings and attributes
+# SRv6 over IS-IS (optionally using BGP and/or BGP L3VPN v4/v6) default settings and attributes
 #
-requires: [ isis ]
-transform_after: [ vlan ]
-locator: 2001:db8:aaaa # 2001:db8::/32 see RFC 6890 – Documentation Set
-address_families: [ ipv4, ipv6 ]
-# protocols: [ isis ] # Could model like this too
-bgp: False
-isis: True
+# Modeled similar to the MPLS module
+#
+config_after: [ vlan, isis, ospf, bgp ]
+transform_after: [ vlan, bgp, vrf ]
+no_propagate: [ locator_pool ]
+
+locator_pool: 2001:db8::/40 # Default address pool for locators, see RFC 6890 – Documentation Set
+igp: [ isis ]               # By default, enable SRv6 over IS-IS
+af:
+  ipv4: True
+  ipv6: True
+bgp: True                   # Enable SRv6 for BGP when available
+
 attributes:
   global:
-    locator: str
-    address_families: { type: list, true_value: [ ipv4, ipv6 ] }
+    af:
+      _list_to_dict: True
+      _alt_types: [ NoneType ]
+      ipv4: bool
+      ipv6: bool
     bgp: bool
-    isis: bool
+    igp: { type: list, valid_values: [ isis, ospf ] }
+    vpn: # BGP VPN v4/v6
+      _alt_types: [ bool, BoxList ]
+      ipv4: { type: list, true_value: [ ibgp ] }
+      ipv6: { type: list, true_value: [ ibgp ] }
+    locator_pool: { type: ipv6, use: subnet_prefix }
+
   node:
-    locator: str
-    address_families:
-    bgp: bool
-    isis: bool
+    af:
+    bgp:
+    igp:
+    locator: { type: ipv6, use: subnet_prefix }
+    vpn:
     transit_only: bool
```

Comment:
This seems to be implemented only on SR-OS. Any plans for similar functionality on FRR?

Reply:
I can add a support column, I looked at https://www.segment-routing.net/images/20250311-srv6-usid-frr-netdev-0x19.pdf but the sid ```show``` commands and static sids aren't available in the current FRR version and I figured we could leave that FFS.

The flag helps optimize resource usage on SR OS, since FRR is pure software it doesn't make much difference currently